asset-transfer-basic/chaincode-external-nano/README.md

```diff
@@ -0,0 +1,78 @@
+
+Modify the `externalBuilders` field in the `core.yaml` file to add the external sample builder which comes with this sample so that the configuration looks something like the following:
+```
+externalBuilders:
+    - name: external-sample-builder
+	  path: ../asset-transfer-basic/chaincode-external-nano/sampleBuilder>
+```
+This update sets the name of the external builder as `external-sample-builder`, and the path of the builder to the scripts provided in this sample. The path must be absolute or relative to where the peer will run so that it can find the builder when installing the chaincode.
+
+Note that the name of the builder does not really matter, the peer will try all the builders in order and use the first one for which the `detect` program or script returns 0 which in our sample is achieve with simply verifying that the `metada.json` file contains `type: "external"`.
+
+## Starting the network
+
+Once the above is done, you can proceed with setting up the Test Network Nano following the instructions to start the network. **HOWEVER** stop before getting into [the section on deploying and running the chaincode](../../test-network-nano-bash/Readme.md#Instructions-for-deploying-and-running-the-basic-asset-transfer-sample-chaincode). Instead come back here and proceed with the following.
+
+## Packaging the chaincode as an external service 
+
+In yet another terminal navigate to `fabric-samples/asset-transfer-basic/chaincode-external-nano` and prepare the chaincode pakage with the following commands.
+
+First, create a `code.tar.gz` archive containing the `connection.json` file:
+```
+tar cfz code.tar.gz connection.json
+```
+
+Then, create the chaincode package, including the `code.tar.gz` file and the supplied `metadata.json` file:
+```
+tar cfz external-chaincode.tgz metadata.json code.tar.gz
+```
+
+The `connection.json` configuration file specifies to the peer how to connect to the external chaincode service. Because you'll be running it locally it is set to localhost.
+
+It is worth noting that because the chaincode will run as an external service, the chaincode itself is not included in the chaincode package that gets installed on the peer. Only the configuration and metadata information is included in the package.
+
+## Installing the chaincode
+
+Once the chaincode package is ready, utilize the peer1 admin to simply install the prepackaged chaincode:
+
+```
+peer lifecycle chaincode install ../asset-transfer-basic/chaincode-external-nano/external-chaincode.tgz
+```
+
+Copy the returned chaincode package ID into an environment variable for use in subsequent commands (your ID may be different):
+
+```
+export CC_PACKAGE_ID=basic_1.0:c70ccad6efd2d7ccb4f68816b62eb038a01dacae3ebcc90b23edab27c2db127d
+```
+
+## Starting the chaincode service
+
+Back in your chaincode service terminal, you now need to build the chaincode with the following command:
+```
+go build
+```
```

Comment:
The `asset-transfer-basic` Java and Node chaincodes are already setup to be able to run in the required 'mode' 
I've got a change to make to go to allow it to run in a simpler manner. 

But that change is stacked up behind other things...

Reply:
Yes, I agree with that. I think it'd be nice to be able to use the "standard" chaincode samples instead of this custom one. I tried and I couldn't seem to provide it with the right environment to make it happy. Maybe we could have a chat about it.

Reply:
This works but is not ideal...   hello, `fabric-cli`, where are you?? 

https://github.com/hyperledger/fabric-samples/blob/main/test-network-k8s/scripts/chaincode.sh#L153

```
function set_chaincode_id() {
  local cc_package=build/chaincode/${CHAINCODE_NAME}.tgz
  cc_sha256=$(shasum -a 256 ${cc_package} | tr -s ' ' | cut -d ' ' -f 1)

  label=$( jq -r '.label' chaincode/${CHAINCODE_NAME}/metadata.json)

  CHAINCODE_ID=${label}:${cc_sha256}
}
```

Reply:
So, I must say I'm rather confused by the fact that it seems that different versions of the chaincode use different environment variables. The Java version wants CORE_CHAINCODE_ID_NAME, while the Typescript version wants CHAINCODE_ID. What gives? Is it just because we are in transition and not all versions are in sync?

Reply:
Wouldn't say we in 'transition' more like everything was written by different people originally. 

We need to get this sorted.

Reply:
If we could agree on one way to go I'll be happy to help with transitioning everything in that direction.